### PR TITLE
Setup bazel-bench pipeline on BazelCI

### PR DESCRIPTION
**Pull request by joeleba**
This script drives the [bazel-bench Nightly](https://buildkite.com/bazel/bazel-bench-nightly) pipeline.

What it does:
1. Collect bazel commits of a particular day.
2. From the list of supported projects (defined in bazel_bench.py), get the platforms for each project.
3. For each (project, platform) pair run bazel_bench with the bazel commits from step (1).
4. Upload the results to BQ.

### CODE REVIEW

**Comment by joeleba** [comment]
@meisterT

File: buildkite/bazel_bench.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+    Args:
+      project_name: a string: the name of the project. e.g. "Bazel".
+
+    Returns:
+      A list of string: the platforms for this project.
+    """
+    http_config = bazelci.DOWNSTREAM_PROJECTS[project_name]["http_config"]
+    configs = bazelci.fetch_configs(http_config, None)
+    tasks = configs["tasks"]
```
Comment (fweikert) [inline buildkite/bazel_bench.py]: some configs still use the legacy field "platforms":

tasks = configs.get("tasks") or configs.get("platforms")

File: buildkite/bazel_bench.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  for project in PROJECTS:
+    for platform in get_platforms(project["name"]):
+      # bazel-bench doesn't support Windows for now.
+      if platform == "windows":
+        continue
+      bazel_bench_commands.append(
+          bazel_bench_for_commits(bazel_commits, platform, project))
+
+  # Print the commands
+  print(yaml.dump({"steps": bazel_bench_commands}))
```
Comment (fweikert) [inline buildkite/bazel_bench.py]: you need to use bazelci.create_step() to get something that can actually run (including agent information etc)

File: buildkite/bazel_bench.py (hunk 1 above)
Comment (joeleba) [reply buildkite/bazel_bench.py]: `fetch_configs` already accounts for the legacy format:
https://github.com/bazelbuild/continuous-integration/blob/master/buildkite/bazelci.py#L573

File: buildkite/bazel_bench.py (hunk 2 above)
Comment (joeleba) [reply buildkite/bazel_bench.py]: Done.

File: buildkite/bazel_bench.py (hunk 1 above)
Comment (fweikert) [reply buildkite/bazel_bench.py]: Yeah, I forgot that I had added that code :D

**Pull request merged**